core/src/com/unciv/logic/automation/unit/UnitAutomation.kt

```diff
@@ -632,8 +632,8 @@ object UnitAutomation {
             .map { BattleDamage.calculateDamageToDefender(MapUnitCombatant(it), cityCombatant) }
             .sum() // City heals 20 per turn
 
-        if (expectedDamagePerTurn < city.health && // If we can take immediately, go for it
-                (expectedDamagePerTurn <= 20 || city.health / (expectedDamagePerTurn-20) > 5)){ // otherwise check if we can take within a couple of turns
+        if (expectedDamagePerTurn < city.health * 4 && // If we can take immediately, go for it
+                (expectedDamagePerTurn <= 5 || city.health / (Math.max(1, expectedDamagePerTurn-20)) > 5)){ // otherwise check if we can take within a couple of turns
```

Comment:
This is incorrect - the city heals 20 per turn, that's the basis for this calculation. If you deal 1 per turn for 5 turns, you won't get the city- it's healing faster than your dealing damage!

Reply:
But I agree that we should have put it in a named val cityHealingPerTurn for context clarity